xscertparser/cmd/acklogparser.py

```diff
@@ -61,29 +63,29 @@ def result_parser(tarfilename, logsubdir):  # pylint: disable=R0914,R0912
 
     # XS-version
     for version in test_conf.getElementsByTagName("global_config"):
-        if 'xs_version' in version.attributes.keys():
+        if 'xs_version' in list(version.attributes.keys()):
             SERVER_DICT['xs_version'] = version.attributes['xs_version'].value
 
     # CPU info and HBA pci-id info
     hba_bus_id_list = []
     for device in test_conf.getElementsByTagName("device"):
-        if 'family' in device.attributes.keys():
+        if 'family' in list(device.attributes.keys()):
             SERVER_DICT['family'] = device.attributes['family'].value
-        if 'stepping' in device.attributes.keys():
+        if 'stepping' in list(device.attributes.keys()):
             SERVER_DICT['stepping'] = device.attributes['stepping'].value
-        if 'model' in device.attributes.keys():
+        if 'model' in list(device.attributes.keys()):
             SERVER_DICT['model'] = device.attributes['model'].value
-        if 'modelname' in device.attributes.keys():
+        if 'modelname' in list(device.attributes.keys()):
             SERVER_DICT['modelname'] = device.attributes['modelname'].value
-        if 'socket_count' in device.attributes.keys():
+        if 'socket_count' in list(device.attributes.keys()):
             SERVER_DICT['sockets'] = device.attributes['socket_count'].value
-        if 'PCI_description' in device.attributes.keys():
+        if 'PCI_description' in list(device.attributes.keys()):
             if device.attributes['PCI_description'].value \
                     not in SERVER_DICT['nics']:
                 SERVER_DICT['nics'].append(
                     device.attributes['PCI_description'].value
                     )
-        if 'device' in device.attributes.keys():
+        if 'device' in list(device.attributes.keys()):
```

Comment:
It's not related to your change to python3.
Not sure, but this line seems weird.
Shouldn't it be ` if 'id' in list(device.attributes.keys()): `

Reply:
Indeed it is weird. What's worse is that it does not do anything! `hba_bus_id_list` is initialized in line 70 and appended to here, but the list is never used.  I have looked back at the history of the project and it has never been used since it was added in 2563a0b879d0 (back then it was called `hbaBusIdList`). 
I have removed the dead code.